monitoring/uss_qualifier/scd/simulator/operational_intents_flight_authorisation_data_generator.py

```diff
@@ -0,0 +1,449 @@
+    ''' A class to generate data for operator registration per EN4709-02 standard. 
+    ''' 
+
+    def __init__(self):        
+        
+        self.registration_number_code_points = '0123456789abcdefghijklmnopqrstuvwxyz'
+
+    def generate_incorrect_operator_registration_number(self, valid_operator_id:str) -> str:
+        ''' Take a valid Operator Registration number per the standard and modify it to make it incorrect / invalid '''
+
+        new_registration_number = valid_operator_id.split('-')[0]
+        final_random_string = ''.join(random.choice(string.ascii_lowercase) for _ in range(3))
+        new_registration_number = new_registration_number +'-'+ final_random_string
+        return new_registration_number
+
+    def generate_operator_registration_number(self, prefix='CHE') -> str:
+        ''' A method to generate the Operator Registration number '''
+        def gen_checksum(raw_id):
+            assert raw_id.isalnum()
+            assert len(raw_id) == 15
+            
+            d = {v: k for k, v in enumerate(list(self.registration_number_code_points))}
+            numeric_base_id = list(map(d.__getitem__, list(raw_id)))
+            # Multiplication factors for each digit depending on its position
+            mult_factors = cycle([2, 1])
+            def partial_sum(number, mult_factor):
+                '''Calculate partial sum ofr a single digit.'''
+                quotient, remainder = divmod(number * mult_factor, 36)
+                return quotient + remainder
+            final_sum = sum(                
+                partial_sum(int(character), mult_factor)
+                for character, mult_factor in zip(numeric_base_id, mult_factors))
+
+            # Calculate control number based on partial sums
+            control_number = -final_sum % 36
+            
+            return list(self.registration_number_code_points)[control_number]
+
+        final_random_string = ''.join(random.choice(string.ascii_lowercase) for _ in range(3))
+        base_id = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(12))
+        checksum = gen_checksum(base_id + final_random_string)
+        reg_num = prefix + base_id + str(checksum) +'-'+ final_random_string
+        return reg_num        
+
+
+class KnownIssuesAcceptableResultFieldGenerator():
+    ''' A class to generate Known Issues and Acceptable results as a result of data processing of the test data provided by the USSP '''
+    
+    def __init__(self, expected_flight_authorisation_processing_result:str, expected_operational_intent_processing_result:str):
+
+        self.expected_flight_authorisation_processing_result = expected_flight_authorisation_processing_result
+        self.expected_operational_intent_processing_result = expected_operational_intent_processing_result
+        self.all_test_results = set([expected_flight_authorisation_processing_result, expected_operational_intent_processing_result])
```

Comment:
This should just be `self.all_test_results = {expected_flight_authorisation_processing_result, expected_operational_intent_processing_result}`

Reply:
Thanks for this, I have now updated the definition